tests/Moq.Tests/ProtectedMockFixture.cs

```diff
@@ -426,13 +450,37 @@ public void ThrowsIfVerifyEmptyResultMethodName()
 		[Fact]
 		public void ThrowsIfVerifyVoidMethodNotFound()
 		{
-			Assert.Throws<ArgumentException>(() => new Mock<FooBase>().Protected().Verify("Foo", Times.Once()));
+			var actual = Record.Exception(() =>
+			{
+				new Mock<FooBase>().Protected().Verify("Foo", Times.Once());
+			});
+
+			Assert.IsType<ArgumentException>(actual);
+			Assert.Equal("No Protected Method FooBase.Foo found whose parameters are compatible with provided types : ().", actual.Message);
 		}
 
 		[Fact]
 		public void ThrowsIfVerifyResultMethodNotFound()
 		{
-			Assert.Throws<ArgumentException>(() => new Mock<FooBase>().Protected().Verify<int>("Foo", Times.Once()));
+			var actual = Record.Exception(() =>
+			{
+				new Mock<FooBase>().Protected().Verify<int>("Foo", Times.Once());
+			});
+
+			Assert.IsType<ArgumentException>(actual);
+			Assert.Equal("No Protected Method FooBase.Foo found whose parameters are compatible with provided types : ().", actual.Message);
+		}
+
+		[Fact]
+		public void ThrowsIfVerifyIncompatibleArgumentSupplied()
+		{
+			var actual = Record.Exception(() =>
+			{
+				new Mock<FooBase>().Protected().Verify<string>("StringArg", Times.Once(), ItExpr.IsAny<int>());
+			});
+
+			Assert.IsType<ArgumentException>(actual);
+			Assert.Equal("No Protected Method FooBase.StringArg found whose parameters are compatible with provided types : (System.Int32).", actual.Message);
```

Comment:
Once the implementation uses `TypeNameFormatter`, you'll need to change `System.Int32` to `int`. :wink: